src/emitter.ts

```diff
@@ -497,8 +497,11 @@ export function emitWebIDl(webidl: Browser.WebIdl, flavor: Flavor) {
     /// Generate the parameters string for function signatures
     function paramsToString(ps: Browser.Param[]) {
         function paramToString(p: Browser.Param) {
+            if (p.type === "Promise" && !Array.isArray(p.subtype)) {
```

Comment:
what does it mean when `Array.isArray(p.subtype)` and why don't we do anything in that case?

Reply:
`p.subtype` being an array means it's `Promise<TypeA, TypeB>`, which will never happen. This check is mainly to not treat `p.subtype` as a union in this block.